sandbox/cgroup/scanning.go

```diff
@@ -167,6 +159,45 @@ func PidsOfSnap(snapInstanceName string) (map[string][]int, error) {
 		}
 		return nil, err
 	}
+	return pathList, nil
+}
+
+// PidsOfSnap returns the association of security tags to PIDs.
+//
+// NOTE: This function returns a reliable result only if the refresh-app-awareness
+// feature was enabled before all processes related to the given snap were started.
+// If the feature wasn't always enabled then only service process are correctly
+// accounted for.
+//
+// The return value is a snapshot of the pids for a given snap, grouped by
+// security tag. The result may be immediately stale as processes fork and
+// exit.
+//
+// Importantly, if the per-snap lock is held while computing the set, then the
+// following guarantee is true: if a security tag is not among the results then
+// no such tag can come into existence while the lock is held.
+//
+// This can be used to classify the activity of a given snap into activity
+// classes, based on the nature of the security tags encountered.
+func PidsOfSnap(snapInstanceName string) (map[string][]int, error) {
+	// pidsByTag maps security tag to a list of pids.
+	pidsByTag := make(map[string][]int)
```

Comment:
move the declaration of `pidsByTag` down before the for loop, where it's used.

Reply:
Done.